gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -569,6 +577,22 @@ public boolean handle(final RoutingContext ctx) {
             }
         }
 
+        // 1. Check if the request method is GET
+        if (request.method() == HttpMethod.GET) {
+            String uri = request.uri();
+            String queryParam = request.getParam("q");
```

Comment:
Using any other query params than `q` in `GET` requests should result in a `400 Bad Request` and not just return all listeners/hooks. Also create tests to verify this.

Reply:
Ok, I will implement this validation

Reply:
Added another remark about this, since it is not yet fixed.

Reply:
Implemented other validations to cover all situations.